### DISCUSSION TITLE
How to apply the patch for a total noob?

### DISCUSSION THREAD

**Discussion by Shoontz** [bangcorrupt/hacktribe#41] [Q&A] [answered]
Please don't laugh guys, but I read the Firmware Patch how too and I'm still totally lost.  I have Electribe 2 Sampler firmware version 2.02 downloaded but I'm not a dev and not quite sure what to do next.  Is this something I can do with Terminal?  Do I need to download Python?  

Sorry, I know these are dumb questions, but I think I'm in over my head on step one.  Hahaha!  Thanks for any help you can give.  Love this project!

**Comment by bangcorrupt** [comment]
The instructions in the wiki are for a Linux system, commands should be typed in a terminal.  You need `bsdiff` and `git` installed.  If  you are installing to a factory synth (grey or blue) you will need `python` installed so you can modify the file header.  `sha256sum` is usually already part of the system.  

[Linux Fundamentals Part 1](https://tryhackme.com/room/linuxfundamentalspart1) is a good place to start.  

For Windows instructions, see [this video](https://www.youtube.com/watch?v=x6al8AWddIg)  by Thomas Herrmann.

**Reply by Shoontz** [reply]
Thank you, thank you, thank you!  This is SUPER helpful!  OK, I feel confident enough to try this now.  :-D

**Comment by bangcorrupt** [comment, marked as answer]
UPDATE 06/08/2023:

Follow the [Hacktribe Tutorial](https://github.com/bangcorrupt/hacktribe-tutorial) to get the patched firmware files.

Alternatively, see [Other ways to patch](https://github.com/bangcorrupt/hacktribe/wiki/How-To#other-ways-to-patch) in the wiki.

**Reply by Shoontz** [reply]
You beautiful genius!

**Reply by Plugwit** [reply]
In the world of coding I'm like a single cell organism but I would like to be able to open up my electribe (blue E2) into a sampler.
Is or will their be a premade Sd card with the fireware already installed and ready to go option available for purchase as I'm just worried about bricking my electribe.......
It's a brilliant hack that I would love to do but just don't have the confidence......please help.

**Reply by bangcorrupt** [reply]
The original firmware is not mine to sell.  The new patching script makes it much easier, see [patch automatically](https://github.com/bangcorrupt/hacktribe/wiki/How-To#patch-automatically) in the wiki. This will apply the patch and check the hash for you, telling you if it was successful.   If you're struggling with specific steps just ask, I'm happy to help. 

If you've never used a terminal before try some tutorials, I like [Linux Fundamentals](https://tryhackme.com/room/linuxfundamentalspart1).  The process is similar on Windows or Mac.  

In a few weeks there will be a GUI patching tool, where you select a file and click a button.

**Reply by Plugwit** [reply]
In a few weeks there will be a GUI patching tool, where you select a file and click a button.(Quote above)

Yes i hope someone makes this happen for the pond life at the bottom of the food chain like me……lol
Thankyou for your reply…….

**Comment by SibirskySnickers** [comment]
When I try, I get this error. What could be wrong?
![hktrb](https://user-images.githubusercontent.com/100016100/155161183-055fb7fb-cc8e-469e-991b-a9aca398abb7.jpg)

**Reply by maks** [reply]
Sorry I'm not really familiar with how the new GUI patching tool works, but it looks like it can't find the SHA checksum file `SYSTEM.VSB.sha` which you can find in this repo in the `hash` folder.

Perhaps someone else in this discussion who has used the GUI tool on windows may know whats gone wrong, as I don't use windows so bit hard for me to say.

**Reply by bangcorrupt** [reply]
If you clone the whole repo everything should be at the correct path.  At some point I'll make a release containing only the necessary files.  

>     git clone --recursive https://github.com/bangcorrupt/hacktribe-editor.git

EDIT: See comment below.

**Reply by jigginz** [reply]
> When I try, I get this error. What could be wrong? ![hktrb](https://user-images.githubusercontent.com/100016100/155161183-055fb7fb-cc8e-469e-991b-a9aca398abb7.jpg)

I am constantly getting the same errorr when manually selecting the original SYSTEM.VSB file and the hacktribe-2.patch. I have always downloaded the whole repo as a zip file.  Anyone else had this problem?

**Reply by bangcorrupt** [reply]
Looks like this is a combination of hardcoded file paths and moving the editor to a separate repo.  If you clone the main `hacktribe` repo, put the firmware file at `hacktribe/SYSTEM.VSB` and copy the gui exe to `hacktribe/scripts/hacktribe-gui.exe` it should work.  (If you download and extract a zip of hacktribe instead of cloning, the top level directory will be called `hacktribe-main`).

**Reply by jigginz** [reply]
"Firmware patched successfully"

Awesome thanks! The executable just needed to be in: hacktribe/scripts 

Looking forward to getting this up and running, great work

**Reply by TonyHC131** [reply]
Thanks, this is very helpful I was about to going mad for this error...

**Reply by bangcorrupt** [reply]
This is fixed in the latest commit.

**Comment by bangcorrupt** [comment]
@untergeekDE  added a [colab notebook tutorial](https://github.com/bangcorrupt/hacktribe/blob/main/scripts/HackTribing_Notebook.ipynb) walking through the patching process.

There is also a [bash script tutorial](https://github.com/bangcorrupt/hacktribe/blob/main/scripts/hacktribe-tutorial.sh).

**Comment by Tazymany** [comment]
Hello, I am French sorry for my writing.
I would like to put back the original firmware, but when I put the system file of the 2.02 update, the electribe gives me a file error.
Thank you for your help
Tazymany

**Reply by bangcorrupt** [reply]
#70 #193

**Reply by Tazymany** [reply]
Thanks I just understood, the problem is that I modified the file with Hacktribe editor. How do I change the factory firmware header? Thank you for your quick response

**Reply by bangcorrupt** [reply]
https://github.com/bangcorrupt/hacktribe/discussions/70#discussioncomment-4557133

**Comment by Nightshark-dev** [comment]
Can somebody help me? I am trying to download Hacktribe on my blue Korg.I Tryed to do so for the past 5 hours. I did everything there is. I tryed every path, downloaded everything, but still it dosent work. The closest i got was with the first path with the GUI, but on every attempt i got error messages sometimes ERROR File not Found ../ SYSTEM.VSB WARNING:Source file not found then after a bit of reading an trying out the next ERROR: Incorrect source file. WARNING: Electribe 2 Sampler firmrware version 2.02 only. i dont know what to do and then I saw the Notebook but then i also read for red Sampler only. i am defeated. Hacktribe has demolished me. Can somebody help me or say me when the Noetebook for the blue Korg will come out. Sorry for my grammar im from Germany.

**Reply by BKLronin** [reply]
There is just one more step and that is modifying the header so that the sampler firmware is disguised as the synth firmware. Also check your path names are right and that you renamed the hacked file to SYSTEM.VSB.

**Reply by bangcorrupt** [reply]
> `ERROR: Incorrect source file. WARNING: Electribe 2 Sampler firmrware version 2.02 only.`

You need the **sampler** firmware, not synth.

Also, see [this comment](https://github.com/bangcorrupt/hacktribe/discussions/41#discussioncomment-4700681).

**Comment by M3nta1** [comment]
Hello. I don't have the three firmware files like in the video tutorial, and I didn't find the GUI installer.exe in the folders shown in the video. Am I looking in the wrong place?

**Reply by tekkrekk** [reply]
i have the same problem, i already did hacktribe on my blue e2 2 years ago, but now after creating a codespace the 3 files dont appear , it does not start unziping somehow.
